### ISSUE TITLE
Ability to start test resources in parallel (concurrently)

### ISSUE THREAD

**Issue by Nunalves** [quarkusio/quarkus#12056]
**Description**
At the moment it is not possible to start test resources in parallel. (`@QuarkusTestResource` annotated test with `QuarkusTestResourceLifecycleManager` implementation classes)

The use case for me, is that i'm using integration test style tests and sometimes i have to start more than one container (eg: oracle DB and keycloak auth server). It would really speed up test execution if docker could start both containers at the same time instead of one at a time.

There is currently no way of doing this, other than using a single test resource that spawns multiple containers. I want to keep using separate test resources (for reuse).

Example:
```java
@QuarkusTestResource(OracleDBTestResource.class)
@QuarkusTestResource(KeycloakTestResource.class)
public class SomeTest {

    @Test
    public void testStuff() {
(...)
```

Each test resource starts a container. OracleDBTestResource starts an oracle database and KeycloakTestResource starts keycloak.
The KeycloakTestResource#start method is only called after the OracleDBTestResource#start method completes. Which means the keycloak container is only started after the oracle container is started and ready.

What i would like is for both test resources to start in parallel. For both start methods to run simultaneously.

**Implementation ideas**
What i think would be cool from a "user" point of view is to mark the test resource as `parallel = true` (another field in the `@QuarkusTestResource` annotation). And keep it false by default for two reasons:
* for people that upgrade to keep the same behaviour as they currently have
* because if different test resources running simultaneously override the same properties (i don't really know why people would want this) when running them concurrently the last to finish will be the one actually overriding (as opposed to the last to start when running them sequentially)

**Zulip discussion that originated this issue**
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/test.20resources.20-.20start.20in.20parallel

**Comment by Nunalves** [comment]
I have already done this. I just had some trouble with getting the annotation default value because the QuarkusTestResource annotation does not seem to be registered in the jandex index for the test classes.

Please someone check my PR and see what i'm talking about. I wanted to use:
```java
                boolean isParallel = annotation.valueWithDefault(index, "parallel").asBoolean();
```
But couldn't. Had to do:
```
                boolean isParallel = false;
                AnnotationValue parallelAnnotationValue = annotation.value("parallel");
                if(parallelAnnotationValue != null){
                    isParallel= true;
                }
```
